.github/workflows/code_quality.yml

```diff
@@ -0,0 +1,31 @@
+name: code quality
+on:
+    pull_request:
+
+jobs:
+    qodana:
+        runs-on: ubuntu-latest
+        timeout-minutes: 120
+
+        steps:
+            - uses: actions/checkout@v2
+
+            - name: Set up JDK 11
+              uses: actions/setup-java@v1
+              with:
+                  java-version: 11
+
+            - name: Download
+              uses: gradle/gradle-build-action@v1
+              with:
+                  arguments: ":resolveDependencies -Pkotlin.incremental=false --no-daemon"
+
+            - name: Build
+              uses: gradle/gradle-build-action@v1
+              with:
+                  arguments: "assemble testClasses -Pkotlin.incremental=false --no-daemon"
```

Comment:
Do we really need to compile code? I thought the main idea of Qodana is to analyze the code without compilation